### PR TITLE
bring PriorityQueue/Heap implementations inline

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

We triggered extra allocations for the generics on PriorityQueue/Heap.
They unfortunately can't be solved using `@_specialize` as the element
type is `ScheduledTask`.
Fortunately we never exposed those types externally so we can just
inline without breaking the public API.

Modifications:

Moved everything from the `NIOPriorityQueue` module into the `NIO`
module.

Result:

Less allocations, more happiness.

### CODE REVIEW

File: Sources/NIO/Embedded.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@
 //===----------------------------------------------------------------------===//
 
 import Dispatch
-import NIOPriorityQueue
+import NIO
```
[the remark this answers is not in the document]
Comment (weissi) [reply Sources/NIO/Embedded.swift]: thanks, we don't, removed

File: Sources/NIO/EventLoop.swift [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 import NIOConcurrencyHelpers
 import Dispatch
-import NIOPriorityQueue
+import NIO
```
[the remark this answers is not in the document]
Comment (weissi) [reply Sources/NIO/EventLoop.swift]: thanks, we don't, removed

**Comment by Lukasa** [comment]
Uh... because of the way we wrote our Package.swift I think this might be breaking.

The SwiftPM docs say that a target is available to other packages if it's a part of a product. This target was: it was part of the NIO product. So arguably we should be delaying this to 2.0.

**Comment by normanmaurer** [comment]
@weissi yeah that sounds ok to me...

**Comment by Lukasa** [comment]
Works for me.

**Comment by weissi** [comment]
done in #250 . Thanks very much @Lukasa for pointing that out!

**Pull request merged**